jupyterlab_caip_optimizer/src/service/optimizer.spec.ts

```diff
@@ -0,0 +1,61 @@
+/**
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { OptimizerService } from './optimizer';
+import { ServerConnection } from '@jupyterlab/services';
+import { MetadataRequired } from '../types';
+import { fakeStudy, fakeStudyResponse, fakeStudyListResponse } from './test-contsants';
+import { asApiResponse } from 'gcp_jupyterlab_shared';
+
+describe('OptimizerService', () => {
+  const mockSubmit = jest.fn();
+  const mockMakeRequest = jest.fn();
+  const optimizerService = new OptimizerService({
+    submit: mockSubmit,
+  });
+
+  const fakeMetadataRequired = {
+    projectId: '1',
+    region: 'us-central1',
+  } as MetadataRequired;
+
+  beforeEach(() => {
+    ServerConnection.makeRequest = mockMakeRequest;
+  });
+
+  it('Creates a study', async () => {
+    mockSubmit.mockReturnValue(asApiResponse(fakeStudyResponse));
```

Comment:
I would make sure to add `expect.assertions` to the top of these asynchronous tests. [Docs](https://jestjs.io/docs/en/expect#expectassertionsnumber). [Why they are important.](https://stackoverflow.com/questions/50816254/necessary-to-use-expect-assertions-if-youre-awaiting-any-async-function-calls)

Reply:
I think they are more relevant in asynchronous tests involving `try... catch`, as in [here](https://github.com/GoogleCloudPlatform/jupyter-extensions/blob/c1fb81d9c22575255dd372991aeda5df0321aed0/jupyterlab_gcpscheduler/src/service/gcp.spec.ts#L98), as opposed to [here](https://github.com/GoogleCloudPlatform/jupyter-extensions/blob/c1fb81d9c22575255dd372991aeda5df0321aed0/jupyterlab_gcpscheduler/src/service/gcp.spec.ts#L66).